src/components/hooks/useWS.ts

```diff
@@ -15,13 +16,29 @@ const useWS = <T extends TSocketEndpointNames>(name: T) => {
     const send = useCallback(
         async (data?: Parameters<typeof apiManager.augmentedSend<T>>[1]) => {
             setIsLoading(true)
-            try {
-                const response = await apiManager.augmentedSend(name, data)
-                setData(response[name] as TSocketResponseData<T>)
-            } catch (e) {
-                setError(e)
-            } finally {
-                setIsLoading(false)
+            const readyState = parseInt(sessionStorage.getItem('websocket_ready_state'))
+            if (readyState === 1 || readyState === 0) {
```

Comment:
What else it can be?

Why you do not consider both undef, and 0 as a candidate for reconnect?

Reply:
while connecting the state will be set to 0 
after connected it will be 1 
so no need to check for undef